engines/osxnative/user.go

```diff
@@ -0,0 +1,149 @@
+package osxnative
+
+import (
+	"github.com/satori/go.uuid"
+	"os"
+	"path"
+	"strconv"
+)
+
+type user struct {
+	d    dscl
+	name string
+}
+
+func (u *user) create() error {
+	uid, err := u.getMaxUid()
+	if err != nil {
+		return err
+	}
+
+	staffGid, err := u.d.read("/Groups/staff", "PrimaryGroupID")
+	if err != nil {
+		u.delete()
+		return err
+	}
+
+	gid, err := strconv.Atoi(staffGid)
+	if err != nil {
+		return err
+	}
+
+	name := "worker-" + uuid.NewV4().String()
+	userPath := path.Join("/Users", name)
+
+	err = u.d.create(userPath)
+	if err != nil {
+		return err
+	}
+
+	// We set the uid, then check if it is unique, if
+	// not, increment and try again
+	duplicated := true
+	for duplicated {
+		uid++
+		strUid := strconv.FormatInt(int64(uid), 10)
+
+		err = u.d.create(userPath, "uid", strUid)
+		if err != nil {
+			u.delete()
+			return err
+		}
+
+		duplicated, err = u.isDuplicatedId("uid", uid)
```

Comment:
I have to admit I'm not quite sure what isDuplicatedId does.  Can you create a user with the same ID as another user?

Also some comments around some of these methods as to the intention of them would help.  

I tried running the 'dscl . list /Users <name>' command for a known user on my system and it returned nothing.  Not sure what I should be expecting.